cds_ils/importer/providers/cds/rules/base.py

```diff
@@ -510,38 +509,48 @@ def urls(self, key, value):
     sub_u = clean_val("u", value, str, req=True)
 
     eitems_ebl = self["_migration"]["eitems_ebl"]
+    eitems_safari = self["_migration"]["eitems_safari"]
     eitems_external = self["_migration"]["eitems_external"]
     eitems_proxy = self["_migration"]["eitems_proxy"]
     eitems_files = self["_migration"]["eitems_file_links"]
 
-    url = {"value": sub_u}
+    eitem = {"url": {"value": sub_u}, "open_access": False}
     if sub_y and sub_y != "ebook":
-        url["description"] = sub_y
+        open_access = "open access" in sub_y.lower()
+        eitem["url"]["description"] = sub_y
+        eitem["open_access"] = open_access
 
     # EBL publisher login required
     if all([elem in sub_u for elem in ["cds", ".cern.ch" "/auth.py"]]):
-        eitems_ebl.append(url)
+        eitem["open_access"] = False
+        eitems_ebl.append(eitem)
         self["_migration"]["eitems_has_ebl"] = True
     # EzProxy links
     elif "ezproxy.cern.ch" in sub_u:
-        url["value"] = url["value"].replace(
+        eitem["url"]["value"] = eitem["url"]["value"].replace(
             "https://ezproxy.cern.ch/login?url=", ""
         )
-        eitems_proxy.append(url)
+        eitems_proxy.append(eitem)
         self["_migration"]["eitems_has_proxy"] = True
+    # Safari links
+    elif sub_u.startswith("https://learning.oreilly.com/library/view/"):
+        eitem["open_access"] = False
+        eitems_safari.append(eitem)
+        self["_migration"]["eitems_has_safari"] = True
     # local files
     elif all(
         [elem in sub_u for elem in ["cds", ".cern.ch/record/", "/files"]]
     ):
-        eitems_files.append(url)
+        del eitem["open_access"]
```

Comment:
could you add a comment for this line? That the open_access is controlled by files restrictions itself?